### ISSUE TITLE
Scanner is too CPU-intensive

### ISSUE THREAD

**Issue by nanawel** [pR0Ps/PhotoFloat#33]
Hello and thanks for your fork of this really good piece of software.

I'm having an issue with the scanner because the thumbnails generation leads to a huge load on my server.

Would it be possible to add a configurable limit on the number of parallel processes that perform this task?
In the meantime, how can I easily cap the used resources? I've tried `cpulimit` but when used to launch the `photofloat` command it does not apply to the children, and running the command for each process is tedious (as I guess they might change over time).

Thanks

**Comment by pR0Ps** [comment]
I've recently heard reports of the memory usage getting out of control with lots of processes from other people too. Currently it's using as many processes as you have CPU cores. On low-RAM, high-core systems this is probably a bad time. Makes sense to allow limiting it with a command line option.

As a quick hack, you can change [this line](https://github.com/pR0Ps/PhotoFloat/blob/b08362fe2c23c75c61e1787ce589625227c99898/scanner/__main__.py#L154)
```python
with multiprocessing.Pool() as p:
```
to
```python
with multiprocessing.Pool(processes=2) as p:
```
(or whatever limit you want)

**Comment by nanawel** [comment]
Hi @pR0Ps and thanks for your quick reply.
Unfortunately it doesn't seem to change anything. I've modified the line as advised, setting even `1` to lower the CPU usage at the maximum:
```python
        with log_depth():
            with multiprocessing.Pool(processes=1) as p:
                p.map(generate_thumbs, uniques)
```
but I still see 4 `python3` processes running and generating a load > 8 when it reaches the thumbnails generation.

Is there any binary cache somewhere that could explain this behavior? (I've deleted any `.pyc` I could find already)
